### PR TITLE
Team header redesign

### PR DESCRIPTION
**Pull request by jakob223**
Also use the nav header on mobile.

image

cc @keybase/y2ksquad

### CODE REVIEW

File: shared/common-adapters/avatar.render.desktop.tsx [hunk 1]
```diff
@@ -86,11 +86,18 @@ const styles = Styles.styleSheetCreate(
         position: 'absolute',
         right: 0,
       },
-      editTeam: {
-        bottom: -2,
-        position: 'absolute',
-        right: -18,
-      },
+      editTeam: Styles.platformStyles({
```
Comment (hyperobject) [inline shared/common-adapters/avatar.render.desktop.tsx]: This doesn't look quite right with the flag off, may need to do some conditional stuff with the flag here?
Screen Shot 2020-02-03 at 4 27 56 PM

File: shared/teams/team/nav-header/index.new.tsx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+//     />
+//     <AddPeopleHow
+//       attachTo={props.getAttachmentRef}
+//       onHidden={props.toggleShowingMenu}
+//       teamID={props.teamID}
+//       visible={props.showingMenu}
+//     />
+//   </>
+// )
+// const AddPeopleButton = Kb.OverlayParentHOC(_AddPeopleButton)
```
Comment (hyperobject) [inline shared/teams/team/nav-header/index.new.tsx]: remove, maybe?

File: shared/teams/team/nav-header/index.new.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+          <Kb.Text type="BodySmall" style={styles.greenText}>
+            Active
+          </Kb.Text>
+        </Kb.Box2>
+      )}
+      <Kb.Box2 direction="horizontal" gap="tiny" alignItems="center" style={styles.rightActionsContainer}>
+        {props.canChat && <Kb.Button label="Chat" onClick={props.onChat} small={true} />}
+        {props.canEdit && <Kb.Button label="Edit" onClick={props.onEdit} small={true} mode="Secondary" />}
+        <Kb.Button label="Share" onClick={props.onShare} small={true} mode="Secondary" />
+        {/* {props.canAddPeople && <AddPeopleButton teamID={props.teamID} />} */}
```
Comment (hyperobject) [inline shared/teams/team/nav-header/index.new.tsx]: 🔥 ?

File: shared/teams/team/nav-header/index.new.tsx [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+            />
+          )}
+          {(!Styles.isMobile || !!props.role) && (
+            <>
+              <Kb.Text type="BodySmall">
+                {Styles.isMobile
+                  ? capitalize(props.role)
+                  : `You are ${roleDisplay[props.role] || 'a member of'} this team. `}
+              </Kb.Text>
+              {props.role === 'none' && (
```
Comment (buoyad) [inline shared/teams/team/nav-header/index.new.tsx]: image
Team name needs ellipsis; it isn't clear that "Add yourself" is a link

File: shared/teams/team/nav-header/index.new.tsx (hunk 2 above)
Comment (jakob223) [reply shared/teams/team/nav-header/index.new.tsx]: fair enough - we have the other file for when making the new component

File: shared/teams/team/nav-header/index.new.tsx (hunk 4 above)
Comment (jakob223) [reply shared/teams/team/nav-header/index.new.tsx]: I'd rather make the team name multiple lines than ellipsize it. We don't have a blessed style for underlining that link; should I just manually style it?

File: shared/common-adapters/avatar.render.desktop.tsx [hunk 5]
```diff
@@ -65,7 +65,12 @@ const Avatar = (props: Props) => {
         />
       )}
       {props.followIconType && <Icon type={props.followIconType} style={props.followIconStyle} />}
-      {props.editable && <Icon type="iconfont-edit" style={props.isTeam ? styles.editTeam : styles.edit} />}
+      {props.editable && (
+        <Icon
+          type="iconfont-edit"
+          style={props.isTeam ? (flags.teamsRedesign ? styles.editTeam : styles.editTeamOld) : styles.edit}
```
Comment (buoyad) [inline shared/common-adapters/avatar.render.desktop.tsx]: Seems `flags` isn't imported in this file.

File: shared/common-adapters/avatar.render.native.tsx [hunk 6]
```diff
@@ -136,11 +137,24 @@ const styles = Styles.styleSheetCreate(
...
       },
-      editTeam: {
-        bottom: -2,
-        position: 'absolute',
-        right: -28,
-      },
+      editTeam: flags.teamsRedesign
+        ? ({
+            backgroundColor: Styles.globalColors.blue,
+            borderColor: tyles.globalColors.white,
```
Comment (buoyad) [inline shared/common-adapters/avatar.render.native.tsx]: `Styles`

File: shared/teams/team/nav-header/index.new.tsx (hunk 4 above)
Comment (buoyad) [reply shared/teams/team/nav-header/index.new.tsx]: Multiline name sgtm. Not sure, I wonder if the blue "add yourself" banner will exist on mobile too. Fine to leave as is for now.

**Pull request merged**